test/array.ts

```diff
@@ -168,3 +168,21 @@ test('array.ofType', t => {
 		ow(['foo', 'b'], 'foo', ow.array.ofType(ow.string.minLength(3)));
 	}, '(array `foo`) Expected string to have a minimum length of `3`, got `b`');
 });
+
+test('array.exactShape', t => {
+	t.notThrows(() => {
+		ow(['🦄', 2, 3, true, {isFirstCommit: true}], ow.array.exactShape([ow.string, ow.number, ow.number, ow.boolean, ow.object.exactShape({isFirstCommit: ow.boolean})]));
+	});
+
+	t.throws(() => {
+		ow(['🦄', 2, 'nope', true, {isFirstCommit: true}], ow.array.exactShape([ow.string, ow.number, ow.number, ow.boolean, ow.object.exactShape({isFirstCommit: ow.string})]));
+	}, 'Expected property `2` to be of type `number` but received type `string` in array');
+
+	t.throws(() => {
+		ow(['🦄', 'nope', {isFirstCommit: true}], ow.array.exactShape([ow.string, ow.string, ow.object.exactShape({isFirstCommit: ow.boolean}), ow.number, ow.boolean]));
+	}, 'Expected property `3` to be of type `number` but received type `undefined` in array');
+
+	t.throws(() => {
+		ow(['🦄', {isFirstCommit: true}, 'nope', 5, {accepted: false}], ow.array.exactShape([ow.string, ow.object.exactShape({isFirstCommit: ow.boolean}), ow.string]));
+	}, 'Did not expect property `3` to exist, got `5` in array');
```

Comment:
I don't think you should reuse the object shape stuff. The error message is sub-optimal. It should talk about array elements, not properties.

Reply:
I have added isArray as an option in exact method declaration to be able to change error message